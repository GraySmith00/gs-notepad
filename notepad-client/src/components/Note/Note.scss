.note {
  display: flex;
  margin-bottom: 2rem;
  color: rgb(66, 71, 74);

  &__tag {
    font-style: italic;
    color: rgb(87, 95, 99);
    margin-top: 0.5rem;
    font-size: 1.4rem;
  }

  &__text {
    margin-left: 2rem;
  }
}
