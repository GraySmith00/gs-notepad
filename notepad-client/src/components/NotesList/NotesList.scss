.notes-list {
  height: 100%;
}
