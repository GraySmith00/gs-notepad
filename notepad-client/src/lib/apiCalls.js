export const postNote = async note => {
  const url = process.env.REACT_APP_POST_NOTE;
  const response = await fetch(url, {
    method: 'POST',
    body: JSON.stringify(note),
    headers: {
      'Content-Type': 'application/json'
    }
  });

  if (response.status > 299) {
    const error = await response.json();
    throw new Error(error.message);
  }

  const data = await response.json();
  return data;
};

export const getNotes = async () => {
  const url = process.env.REACT_APP_GET_NOTES;
  const response = await fetch(url);
  const data = await response.json();
  return data;
};
