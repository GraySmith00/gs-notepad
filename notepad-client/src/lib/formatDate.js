import moment from 'moment';

export const formatDate = timeStamp => {
  return moment(timeStamp).format('L');
};
