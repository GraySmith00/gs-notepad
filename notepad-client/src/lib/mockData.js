export const mockNotes = [
  {
    createdAt: 1545951483012,
    text: 'buy groceries',
    tag: 'hobby',
    id: 'dd5aa440-0a2a-11e9-93fa-a7949507e226'
  },
  {
    createdAt: 1545951433152,
    text: 'do laundry',
    tag: 'work',
    id: 'bfa29c00-0a2a-11e9-93fa-a7949507e226'
  },
  {
    createdAt: 1545951460092,
    text: 'play golf',
    tag: 'hobby',
    id: 'cfb153c0-0a2a-11e9-93fa-a7949507e226'
  },
  {
    createdAt: 1545951504338,
    text: 'buy gifts',
    tag: 'personal',
    id: 'ea10bb20-0a2a-11e9-93fa-a7949507e226'
  },
  {
    createdAt: 1545451632475,
    text: 'set up tv',
    tag: 'personal',
    id: '425ca4b0-0a2b-11e9-93fa-a7949507e226'
  },
  {
    createdAt: 1545951440518,
    text: 'run errands',
    tag: 'work',
    id: 'c4069260-0a2a-11e9-93fa-a7949507e226'
  }
];
