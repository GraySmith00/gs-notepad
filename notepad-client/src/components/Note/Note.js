import React from 'react';
import PropTypes from 'prop-types';
import { formatDate } from '../../lib/formatDate';

const Note = ({ text, createdAt, tag }) => {
  return (
    <div className="note">
      <h3>{text}</h3>
      <p>Time: {formatDate(createdAt)}</p>
      <p>Tag: {tag}</p>
    </div>
  );
};

Note.propTypes = {
  text: PropTypes.string.isRequired,
  createdAt: PropTypes.number.isRequired,
  tag: PropTypes.string.isRequired
};

export default Note;
