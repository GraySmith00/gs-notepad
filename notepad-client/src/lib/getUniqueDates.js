import { formatDate } from './formatDate';

export const getUniqueDates = notes => {
  const dates = notes.reduce((dates, note) => {
    const formattedDate = formatDate(note.createdAt);

    if (!dates[formattedDate]) {
      dates[formattedDate] = 1;
    }

    return dates;
  }, {});

  return Object.keys(dates);
};
