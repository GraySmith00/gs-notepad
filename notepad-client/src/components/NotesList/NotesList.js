import React from 'react';

import Note from '../Note';

const NotesList = ({ notes }) => {
  return (
    <div className="notes-list">
      {notes.map(note => (
        <Note key={note.id} {...note} />
      ))}
    </div>
  );
};

export default NotesList;
