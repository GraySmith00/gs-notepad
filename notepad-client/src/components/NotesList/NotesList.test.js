import React from 'react';
import { shallow } from 'enzyme';
import { mockNotes } from '../../lib/mockData';
import NotesList from './NotesList';
import Note from '../Note';

describe('NotesList component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <NotesList notes={mockNotes} tagFilter="work" dateFilter={null} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should filter the notes if there is a valid tagFilter', () => {
    wrapper = shallow(
      <NotesList notes={mockNotes} tagFilter="work" dateFilter={null} />
    );

    expect(wrapper.children().find(Note).length).toEqual(2);
  });

  it('should filter the notes if there is a valid dateFilter', () => {
    wrapper = shallow(
      <NotesList notes={mockNotes} tagFilter={null} dateFilter={'12/21/2018'} />
    );

    expect(wrapper.children().find(Note).length).toEqual(1);
  });
});
