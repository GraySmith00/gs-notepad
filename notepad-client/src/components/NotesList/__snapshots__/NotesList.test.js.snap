// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesList component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotesList
    dateFilter={null}
    notes={
      Array [
        Object {
          "createdAt": 1545951483012,
          "id": "dd5aa440-0a2a-11e9-93fa-a7949507e226",
          "tag": "hobby",
          "text": "buy groceries",
        },
        Object {
          "createdAt": 1545951433152,
          "id": "bfa29c00-0a2a-11e9-93fa-a7949507e226",
          "tag": "work",
          "text": "do laundry",
        },
        Object {
          "createdAt": 1545951460092,
          "id": "cfb153c0-0a2a-11e9-93fa-a7949507e226",
          "tag": "hobby",
          "text": "play golf",
        },
        Object {
          "createdAt": 1545951504338,
          "id": "ea10bb20-0a2a-11e9-93fa-a7949507e226",
          "tag": "personal",
          "text": "buy gifts",
        },
        Object {
          "createdAt": 1545451632475,
          "id": "425ca4b0-0a2b-11e9-93fa-a7949507e226",
          "tag": "personal",
          "text": "set up tv",
        },
        Object {
          "createdAt": 1545951440518,
          "id": "c4069260-0a2a-11e9-93fa-a7949507e226",
          "tag": "work",
          "text": "run errands",
        },
      ]
    }
    tagFilter="work"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Note
          createdAt={1545951433152}
          id="bfa29c00-0a2a-11e9-93fa-a7949507e226"
          tag="work"
          text="do laundry"
        />,
        <Note
          createdAt={1545951440518}
          id="c4069260-0a2a-11e9-93fa-a7949507e226"
          tag="work"
          text="run errands"
        />,
      ],
      "className": "notes-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "bfa29c00-0a2a-11e9-93fa-a7949507e226",
        "nodeType": "function",
        "props": Object {
          "createdAt": 1545951433152,
          "id": "bfa29c00-0a2a-11e9-93fa-a7949507e226",
          "tag": "work",
          "text": "do laundry",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "c4069260-0a2a-11e9-93fa-a7949507e226",
        "nodeType": "function",
        "props": Object {
          "createdAt": 1545951440518,
          "id": "c4069260-0a2a-11e9-93fa-a7949507e226",
          "tag": "work",
          "text": "run errands",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Note
            createdAt={1545951433152}
            id="bfa29c00-0a2a-11e9-93fa-a7949507e226"
            tag="work"
            text="do laundry"
          />,
          <Note
            createdAt={1545951440518}
            id="c4069260-0a2a-11e9-93fa-a7949507e226"
            tag="work"
            text="run errands"
          />,
        ],
        "className": "notes-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "bfa29c00-0a2a-11e9-93fa-a7949507e226",
          "nodeType": "function",
          "props": Object {
            "createdAt": 1545951433152,
            "id": "bfa29c00-0a2a-11e9-93fa-a7949507e226",
            "tag": "work",
            "text": "do laundry",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "c4069260-0a2a-11e9-93fa-a7949507e226",
          "nodeType": "function",
          "props": Object {
            "createdAt": 1545951440518,
            "id": "c4069260-0a2a-11e9-93fa-a7949507e226",
            "tag": "work",
            "text": "run errands",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
